mantidproject/mantid
# Reset global variables in python scripts when running entire script

**Description of work.**

This PR resets the global variable list in workbench python interpreter when running a whole script, this should not reset the variables if only running a partial script.

**Report to:** James Lord

**To test:**

1. Run the following python script
```
x=5
print(x)
```
2. remove the `x=5` line and highlight the print statment and run only that line. the x varable should still be kept.
3. Run the whole script without the `x=5` line, it should reset the varables so `x` will not be assigned.

Fixes #27185


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
